widgetssdk/src/main/java/com/glia/widgets/core/survey/GliaSurveyRepository.java

```diff
@@ -0,0 +1,54 @@
+package com.glia.widgets.core.survey;
+
+import androidx.annotation.NonNull;
+
+import com.glia.androidsdk.Engagement;
+import com.glia.androidsdk.GliaException;
+import com.glia.androidsdk.RequestCallback;
+import com.glia.androidsdk.engagement.Survey;
+import com.glia.widgets.di.Dependencies;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class GliaSurveyRepository implements RequestCallback<Survey> {
+
+    private Engagement engagement;
+    private final List<OnSurveyListener> listeners = new ArrayList<>();
+
+    public void onEngagementEnded(Engagement engagement) {
+        if (listeners.isEmpty()) { // No action needs if the listeners are empty.
+            return;
+        }
+        if (this.engagement != null) { // Prevent multiple surveys fetching.
+            return;
+        }
+        this.engagement = engagement;
+        engagement.getSurvey(this);
+    }
+
+    @Override
+    public void onResult(Survey survey, GliaException e) {
+        List<OnSurveyListener> listenersCopy = new ArrayList<>(listeners);
+        for (OnSurveyListener listener : listenersCopy) {
+            listener.onSurveyLoaded(survey);
+        }
+        engagement = null;
+    }
+
+    public void registerListener(OnSurveyListener listener) {
+        listeners.add(listener);
+    }
+
+    public void unregisterListener(OnSurveyListener listener) {
+        listeners.remove(listener);
+    }
+
+    public void submitSurveyAnswers(@NonNull List<Survey.Answer> answers,
+                                    @NonNull String surveyId,
+                                    @NonNull String engagementId,
+                                    @NonNull Consumer<GliaException> callback) {
+        Dependencies.glia().submitSurveyAnswers(answers, surveyId, engagementId, callback);
```

Comment:
Could pass this interface on Constructor and hold reference to it instead :) 

@jakob-mc we might need to create some sort of debt task to pass Glia interface to Repositories on class creation as Glia team introduced Core SDK interface

Reply:
Not sure if its worth doing here tho